src/app/shared/http-services/progress-csv.service.ts

```diff
@@ -0,0 +1,20 @@
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { appConfig } from '../helpers/config';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ProgressCSVService {
+  constructor(private http: HttpClient) {}
+
+  getCSVData(groupId: string, type: 'group' | 'team' | 'user', parentItemIds: string[]): Observable<string> {
+    const params = new HttpParams().set('parent_item_ids', parentItemIds.join(','));
+    return this.http
+      .get(`${appConfig.apiUrl}/groups/${ groupId }/${ type }-progress-csv`, {
+        params: params,
+        responseType: 'text',
+      });
+  }
+}
```

Comment:
When testing this acceptance criteria of the issue #641: 
> The user does not leave the SPA when downloading the csv

I was able to leave the app. Can you check ?

I tested this behavior by throttling the internet connection in the devtools (preset `slow 3G` in chromium), then click on the `export` button and close the tab (with CTRL+W) before having a response from the API

Reply:
My misunderstanding, my bad. Resolving it.